### DISCUSSION TITLE
ldif_read_file: Permission denied for "/etc/openldap/slapd.d/cn=config.ldif". I don't know what exactly is the cause of this problem, can anyone help me？I would be very grateful if anyone could help me.

### DISCUSSION THREAD

**Discussion by AstridTrj** [dcm4che/dcm4chee-arc-light#3779] [General]
[+] Running 4/4
 ⠿ Network dcm4chee_default   Created                                                                                                                                                                                                       0.1s
 ⠿ Container dcm4chee-db-1    Created                                                                                                                                                                                                       0.1s
 ⠿ Container dcm4chee-ldap-1  Created                                                                                                                                                                                                       0.1s
 ⠿ Container dcm4chee-arc-1   Created                                                                                                                                                                                                       0.1s
Attaching to dcm4chee-arc-1, dcm4chee-db-1, dcm4chee-ldap-1
dcm4chee-ldap-1  | 62ff007e @(#) $OpenLDAP: slapd 2.4.48 (May  7 2020 12:49:13) $
dcm4chee-ldap-1  | 	openldap
dcm4chee-ldap-1  | 62ff007e ldif_read_file: Permission denied for "/etc/openldap/slapd.d/cn=config.ldif"
dcm4chee-ldap-1  | 62ff007e slapd stopped.
dcm4chee-ldap-1  | 62ff007e connections_destroy: nothing to destroy.
dcm4chee-db-1    | 
dcm4chee-db-1    | PostgreSQL Database directory appears to contain a database; Skipping initialization
dcm4chee-db-1    | 
dcm4chee-db-1    | 2022-08-19 11:16:14.946 CST [1] LOG:  starting PostgreSQL 12.2 (Debian 12.2-2.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
dcm4chee-db-1    | 2022-08-19 11:16:14.946 CST [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
dcm4chee-db-1    | 2022-08-19 11:16:14.946 CST [1] LOG:  listening on IPv6 address "::", port 5432
dcm4chee-db-1    | 2022-08-19 11:16:14.947 CST [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
dcm4chee-db-1    | 2022-08-19 11:16:14.966 CST [28] LOG:  database system was interrupted; last known up at 2022-08-18 16:20:41 CST
dcm4chee-ldap-1 exited with code 1
dcm4chee-db-1    | 2022-08-19 11:16:15.165 CST [28] LOG:  database system was not properly shut down; automatic recovery in progress
dcm4chee-db-1    | 2022-08-19 11:16:15.167 CST [28] LOG:  redo starts at 0/188FE98
dcm4chee-db-1    | 2022-08-19 11:16:15.167 CST [28] LOG:  invalid record length at 0/188FF80: wanted 24, got 0
dcm4chee-db-1    | 2022-08-19 11:16:15.167 CST [28] LOG:  redo done at 0/188FF48
dcm4chee-db-1    | 2022-08-19 11:16:15.178 CST [1] LOG:  database system is ready to accept connections
dcm4chee-arc-1   | Waiting for ldap:389 ...
dcm4chee-arc-1   | nc: getaddrinfo for host "ldap" port 389: Name or service not known
dcm4chee-arc-1   | nc: getaddrinfo for host "ldap" port 389: Name or service not known
dcm4chee-arc-1   | nc: getaddrinfo for host "ldap" port 389: Name or service not known
dcm4chee-arc-1   | nc: getaddrinfo for host "ldap" port 389: Name or service not known
dcm4chee-arc-1   | nc: getaddrinfo for host "ldap" port 389: Name or service not known
dcm4chee-arc-1   | nc: getaddrinfo for host "ldap" port 389: Name or service not known
dcm4chee-arc-1   | nc: getaddrinfo for host "ldap" port 389: Name or service not known

**Comment by vrindanayak** [comment]
@AstridTrj : Can you share your docker files - how have you started the services?

**Reply by AstridTrj** [reply]
> @AstridTrj : Can you share your docker files - how have you started the services?
This is my docker-compose.yml file
```version: "3"
services:
  ldap:
    image: dcm4che/slapd-dcm4chee:2.4.48-22.2
    logging:
      driver: json-file
      options:
        max-size: "10m"
    ports:
      - "389:389"
    env_file: docker-compose.env
    volumes:
      - /data_warehouse/dcm4chee-arc/ldap:/var/lib/openldap/openldap-data
      - /data_warehouse/dcm4chee-arc/slapd.d:/etc/openldap/slapd.d
    tty: true
  db:
    image: dcm4che/postgres-dcm4chee:12.2-22
    logging:
      driver: json-file
      options:
        max-size: "10m"
    ports:
      - "5432:5432"
    env_file: docker-compose.env
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /etc/timezone:/etc/timezone:ro
      - /data_warehouse/dcm4chee-arc/db:/var/lib/postgresql/data
  arc:
    image: dcm4che/dcm4chee-arc-psql:5.22.2
    logging:
      driver: json-file
      options:
        max-size: "10m"
    ports:
      - "7080:8080"
      - "8443:8443"
      - "9990:9990"
      - "9993:9993"
      - "11112:11112"
      - "2762:2762"
      - "2575:2575"
      - "12575:12575"
    env_file: docker-compose.env
    environment:
      WILDFLY_CHOWN: /opt/wildfly/standalone /storage
      WILDFLY_WAIT_FOR: ldap:389 db:5432
    depends_on:
      - ldap
      - db
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /etc/timezone:/etc/timezone:ro
      - /data_warehouse/dcm4chee-arc/wildfly:/opt/wildfly/standalone
      - /data_warehouse/dcm4chee-arc/storage:/storage
```

docker-compose.env
```
STORAGE_DIR=/storage/fs1
POSTGRES_DB=pacsdb
POSTGRES_USER=pacs
POSTGRES_PASSWORD=pacs
```

I use docker-compose -p dcm4chee up to to pull up the container（I have built the corresponding mirror），and then report the above errors.

**Comment by gunterze** [comment]
Verify if the file owner of your mapped out LDAP directories
```yaml
    volumes:
      - /data_warehouse/dcm4chee-arc/ldap:/var/lib/openldap/openldap-data
      - /data_warehouse/dcm4chee-arc/slapd.d:/etc/openldap/slapd.d
```
matches the user/group ID running slapd in the container, 1021/1021 by default, s.

### [`LDAP_USER_ID`](https://github.com/dcm4che-dockerfiles/slapd-dcm4chee#ldap_user_id)

`uid` of the ldap user running `slapd` inside the container (optional, default is `1021`).

### [`LDAP_GROUP_ID`](https://github.com/dcm4che-dockerfiles/slapd-dcm4chee#ldap_group_id)

`gid` of the ldap user running `slapd` inside the container (optional, default is `1021`).
